src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1678,6 +1678,13 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
 
             if (buildIntegratedProjectsToUpdate.Count != 0)
             {
+                // Only automatically create Source Mappings when there's exclusively BuildIntegratedProjects.
```

Comment:
This doesn't make me feel warm and fuzzy. Maybe have a bolean that checks when something is incompatible and maybe passes a null for new mapping id/new source there, instead of nulling it out here. 

idk if that's any better, but just an idea.

Reply:
Yeah. I think we should add a tracking issue, if there is one.

Reply:
https://github.com/NuGet/Home/issues/12832